src/Mod/Material/Gui/DlgInspectAppearance.cpp

```diff
@@ -0,0 +1,273 @@
+        Gui::ViewProvider* view =
+            Gui::Application::Instance->getDocument(it.pDoc)->getViewProvider(it.pObject);
+        views.push_back(view);
+    }
+
+    return views;
+}
+
+/// @cond DOXERR
+void DlgInspectAppearance::OnChange(Gui::SelectionSingleton::SubjectType& rCaller,
+                                    Gui::SelectionSingleton::MessageType Reason)
+{
+    Q_UNUSED(rCaller);
+
+    if (Reason.Type == Gui::SelectionChanges::AddSelection
+        || Reason.Type == Gui::SelectionChanges::RmvSelection
+        || Reason.Type == Gui::SelectionChanges::SetSelection
+        || Reason.Type == Gui::SelectionChanges::ClrSelection) {
+        std::vector<Gui::ViewProvider*> views = getSelection();
+        update(views);
+    }
+}
+/// @endcond
+
+void DlgInspectAppearance::update(std::vector<Gui::ViewProvider*>& views)
+{
+    App::Document* doc = App::GetApplication().getActiveDocument();
+    if (doc) {
+        ui->editDocument->setText(QString::fromUtf8(doc->Label.getValue()));
+
+        if (views.size() == 1) {
+            auto view = dynamic_cast<Gui::ViewProviderDocumentObject*>(views[0]);
+            if (!view) {
+                return;
+            }
+            auto* obj = view->getObject();
+            if (!obj) {
+                return;
+            }
+            auto* labelProp = dynamic_cast<App::PropertyString*>(obj->getPropertyByName("Label"));
+            if (labelProp) {
+                ui->editObjectLabel->setText(QString::fromUtf8(labelProp->getValue()));
+            }
+            else {
+                ui->editObjectLabel->setText(QLatin1String(""));
+            }
+            ui->editObjectName->setText(QLatin1String(obj->getNameInDocument()));
+
+            auto subElement = Gui::Selection().getSelectionEx();
+            if (subElement.size() > 0) {
+                auto& subObject = subElement[0];
+                if (subObject.getSubNames().size() > 0) {
+                    ui->editSubShape->setText(QString::fromStdString(subObject.getSubNames()[0]));
+                }
+                else {
+                    ui->editSubShape->setText(QLatin1String(""));
+                }
+            }
+            else {
+                ui->editSubShape->setText(QLatin1String(""));
+            }
+
+            auto subShapeType = QString::fromUtf8(obj->getTypeId().getName());
+            subShapeType.remove(subShapeType.indexOf(QLatin1String("::")), subShapeType.size());
+            ui->editSubShapeType->setText(subShapeType);
+            ui->editShapeType->setText(QString::fromUtf8(obj->getTypeId().getName()));
+
+            ui->tabAppearance->clear();
+            if (labelProp && QString::fromUtf8(labelProp->getValue()).size() > 0) {
+                auto* prop =
+                    dynamic_cast<App::PropertyMaterialList*>(view->getPropertyByName("ShapeAppearance"));
+                if (prop) {
+                    for (int index = 0; index < prop->getSize(); index++) {
```

Comment:
@davesrocketshop 
index is incremented twice. I don't think this is the intended behaviour.

Reply:
See #14366